Makefile

```diff
@@ -1,23 +1,35 @@
+# Whether to turn compiler warnings into errors
+export WERROR ?= true
+
 default: release
 
 release:
-	mkdir -p build && cd build && cmake ../ -DCMAKE_BUILD_TYPE=Release && VERBOSE=1 cmake --build .
+	mkdir -p build/Release && cd build/Release && cmake ../../ -DCMAKE_BUILD_TYPE=Release -DWERROR=$(WERROR) && VERBOSE=1 cmake --build .
 
 debug:
-	mkdir -p build && cd build && cmake ../ -DCMAKE_BUILD_TYPE=Debug && VERBOSE=1 cmake --build .
+	mkdir -p build/Debug && cd build/Debug && cmake ../../ -DCMAKE_BUILD_TYPE=Debug -DWERROR=$(WERROR) && VERBOSE=1 cmake --build .
+
+test: release debug
```

Comment:
I didnt merge the fancy if logic from master that checks for the existence of the unit-tests binary since we're compiling both debug and release during tests.